box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxSession.java

```diff
@@ -848,7 +848,7 @@ public void run() {
                     }
                 });
                 try {
-                    while(!mIsWaitingForLoginUi) {
+                    while(mIsWaitingForLoginUi) {
```

Comment:
wait should be called within a synchronized blocked

Reply:
and it is... please expand the code and you'll see it --> "synchronized(mSession)"